packages/frontend/src/utils/transactions.ts

```diff
@@ -7,3 +7,14 @@ export const sortByRecentTimestamp = (txs: Transaction[]) => {
 export function filterByHash(txs: Transaction[] = [], hash: string = '') {
   return txs.filter(tx => tx.hash !== hash)
 }
+
+export async function queryFilterTransferFromL1CompletedEvents(bridge, networkName) {
+  const destL2Bridge = await bridge.getL2Bridge(networkName)
+  const bln = await destL2Bridge.provider.getBlockNumber()
+  const evs = await destL2Bridge.queryFilter(
+    destL2Bridge.filters.TransferFromL1Completed(),
+    bln - 9999,
```

Comment:
We can't assume that a 10k block range will always work because certain rpc endpoints only allow range look ups of `1k` blocks, so pagination would have to happen here